lib/net.js

```diff
@@ -583,33 +581,33 @@ Socket.prototype._destroy = function(exception, cb) {
     clearTimeout(s[kTimeout]);
   }
 
-  debug('close');
+  const onClosed = () => {
```

Comment:
I didn't test but I think this still does not prevent the `'error'` event from being emitted multiple times. Assume that `socket.destroy(err)` is called twice.

First call will wait for this callback to be called and emit `'error'` and `'close'` when that happens.
Second call will find `_readableState.destroyed` set to `true` but `_writableState.errorEmitted` is not set to `true` yet as that will happen when this callback if invoked, so another `'error'` event will be emitted.

Reply:
But isn't that true for any stream? If you call destroy twice with error it will emit error twice... I would say the problem is in the destroy impl then?

Reply:
No, it's emitted only once

```js
const { Writable } = require('stream');

const w = new Writable();

w.on('error', console.log);
w.destroy(new Error('Oops'));
w.destroy(new Error('Oops'));
```

Reply:
The callback on destroy is not a public api. It is used only internally.

Reply:
I'm confused... https://nodejs.org/dist/latest-v10.x/docs/api/stream.html#stream_constructor_new_stream_writable_options

Reply:
You are right, sorry, I was referring to https://nodejs.org/api/stream.html#stream_writable_destroy_error.

Reply:
(We should fix that error being emitted more than once in streams imo.)